src/main/scala/chisel3/util/experimental/decode/DecoderBundle.scala

```diff
@@ -71,15 +75,19 @@ class DecodeBundle(fields: Seq[DecodeField[_, _]]) extends Record {
   * @param fields   all fields as decoded output
   * @tparam I concrete type of input patterns trait
   */
-class DecodeTable[I <: DecodePattern](patterns: Seq[I], fields: Seq[DecodeField[I, _]]) {
+class DecodeTable[I <: DecodePattern](patterns: Seq[I], fields: Seq[DecodeField[I, _ <: Data]]) {
   require(patterns.map(_.bitPat.getWidth).distinct.size == 1, "All instructions must have the same width")
 
   def bundle: DecodeBundle = new DecodeBundle(fields)
 
   private val _table = patterns.map { op =>
     op.bitPat -> fields.reverse.map(field => field.genTable(op)).reduce(_ ## _)
   }
-  val table: TruthTable = TruthTable(_table, BitPat.dontCare(_table.head._2.getWidth))
+
+  val table: TruthTable = TruthTable(
+    patterns.map { op => op.bitPat -> fields.reverse.map(field => field.genTable(op)).reduce(_ ## _) },
```

Comment:
```suggestion
    _table,
```
Is this not still `_table` above?